nasa/harmony-gdal-adapter
# General code clean-up in transform.py.

## Description

This PR primarily covers some linting cleanup in `gdal_subsetter/transform.py`. It's not everything, and not meant to be exhaustive, but I had some time today, and wanted to make a start.

The main things covered in this PR:

* Updates to the `checkstackable` class method (now `is_stackable`), including addition of unit tests for it.
* Augmentation of regular expression to find matching `SUBDATASET` output from `gdalinfo` (I'll pop a comment on that to explain it).
* Adoption of `os.path.join()` when concatenating file paths.
* Standardising the use of either double or single quotation marks. PEP8 encourages the use of one or the other, not both. This code had a lot of both. I went with my personal preference of single quotes.
* Adopting f-strings instead of older style "%s" or `str.format()` syntaxes.
* Ensuring the correct amount of whitespace around operators, commas, etc.
* Changing a couple of variable names here and there.
* Some other minor tweaks here and there, probably more than I should have done in one PR.

Note - when HGA v1.2.6 is deployed, the HGA regression tests will need to be updated by merging [this PR](https://github.com/nasa/harmony-regression-tests/pull/42).

## Jira Issue ID

Long-IP, currently doesn't have a ticket.

## Local Test Steps

* Pull this branch locally.
* Build HGA with `bin/build-image`.
* Pull branch for the [accompanying PR for the HGA regression tests](https://github.com/nasa/harmony-regression-tests/pull/42).
* Run those. They should all pass.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~Documentation updated (if needed).~~
